crates/sncast/src/helpers/scarb_utils.rs

```diff
@@ -63,43 +63,59 @@ pub fn get_scarb_metadata_with_deps(manifest_path: &Utf8PathBuf) -> Result<Metad
     execute_scarb_metadata_command(&command)
 }
 
-#[must_use]
-pub fn verify_or_determine_scarb_manifest_path(
+pub fn assert_manifest_path_exists(
     path_to_scarb_toml: &Option<Utf8PathBuf>,
-) -> Option<Utf8PathBuf> {
+) -> Result<Utf8PathBuf> {
     if let Some(path) = path_to_scarb_toml {
         assert!(path.exists(), "Failed to locate file at path = {path}");
     }
 
     let manifest_path = path_to_scarb_toml.clone().unwrap_or_else(|| {
-        get_scarb_manifest()
-            .context("Failed to obtain manifest path from scarb")
-            .unwrap()
+        get_scarb_manifest().expect("Failed to obtain manifest path from scarb")
     });
 
     if !manifest_path.exists() {
-        return None;
+        return Err(anyhow!(
+            "Path to Scarb.toml manifest does not exist = {manifest_path}"
+        ));
     }
 
-    Some(manifest_path)
+    Ok(manifest_path)
 }
 
-pub fn get_package_metadata<'a>(
+fn get_package_metadata_by_name<'a>(
```

Comment:
This is prob `get_workspace_member_metadata_by_name`, right? It should use `metadata.workspace. members`

Reply:
yup, as we discussed offline - this will be a part of scarb_utils cleanup mentioned in pr description :+1: